### DISCUSSION TITLE
Cannot read bucket location. S3 (Credentials from AWS Command Line Interface)

### DISCUSSION THREAD

**Discussion by gfody** [iterate-ch/cyberduck#15691] [Q&A]
I'm using the profile `S3 (Credentials from AWS Command Line Interface)`
![image](https://github.com/iterate-ch/cyberduck/assets/1146032/ba607a51-c4d0-46b2-afa6-968f42312a02)

**Describe the bug**
Randomly it stops working and throws `Cannot read bucket location`
![image](https://github.com/iterate-ch/cyberduck/assets/1146032/5649e1a2-18ea-413e-8794-80dc63acabfb)

it seems to happen after the sso session has expired or before I've logged in, sometimes it works perfectly after running `aws sso login` sometimes it throws this error, I haven't been able to find any correlation, if I wait a while and try again it works. sometimes trying after 5 minutes works, sometimes it doesn't work for an hour or more.

**Desktop (please complete the following information):**
 - OS: Windows 10 22H2 (Build 19045.4046)
 - Version 8.7.3 (40979)

**Comment by tatze96** [comment]
Can you please [enable debug logging](https://docs.cyberduck.io/cyberduck/support/#debug-logging-level), recreate the issue, and send us the log file afterward? If you don't like to see sensitive data to be shared publicly, please send the log file with the ticket reference GH-15691 to support@cyberduck.io, otherwise attach it here.

**Sidenote**: If you want to share the log file here, make sure to attach it through the GitHub web interface, as it wouldn't be visible to us otherwise.

**Reply by gfody** [reply]
I enabled debug logging and reproduced the issue before and after running `aws sso login` - it has not been able to connect successfully in this session. log file attached, confirmed there's nothing sensitive inside.
[cyberduck.log.gz](https://github.com/iterate-ch/cyberduck/files/14625448/cyberduck.log.gz)

I noticed a couple things in the debug log
- Cyberduck passed "default" as the access key to AWS and got "The AWS Access Key Id you provided does not exist in our records."
- Cyberduck read an expired session token from `~.aws/cli/cache` (the current token appears to be located at `~.aws/sso/cache`)

**Reply by gfody** [reply]
if I manually update the latest json file under `~.aws/cli/cache/` then Cyberduck connects successfully
